darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1892,6 +1923,94 @@ def _is_probabilistic(self) -> bool:
             else True  # all torch models can be probabilistic (via Dropout)
         )
 
+    def _load_encoders(
+        self, tfm_save: "TorchForecastingModel", load_encoders: bool
+    ) -> Tuple[SequentialEncoder, Dict]:
+        """Return the encoders from a model save with several sanity checks."""
+        if self.add_encoders is None:
+            same_encoders = True
+            same_transformer = True
+        elif tfm_save.add_encoders is None:
+            same_encoders = False
+            same_transformer = False
+        else:
+            # transformers are equal if they are instances of the same class
+            self_transformer = self.add_encoders.get("transformer", None)
+            tfm_transformer = tfm_save.add_encoders.get("transformer", None)
+            same_transformer = type(self_transformer) == type(tfm_transformer)
+
+            # encoders are equal if they have the same entries (transformer excluded)
+            self_encoders = {
+                k: v for k, v in self.add_encoders.items() if k != "transformer"
+            }
+            tfm_encoders = {
+                k: v for k, v in tfm_save.add_encoders.items() if k != "transformer"
+            }
+            same_encoders = self_encoders == tfm_encoders
+
+        if load_encoders:
+            # avoid silently overwriting new encoders
+            raise_if_not(
+                same_transformer,
+                f"Transformers defined in the loaded encoders and the new model must have the same type, received "
+                f"({None if tfm_save.add_encoders is None else type(tfm_save.add_encoders.get('transformer', None))}) "
+                f"and "
+                f"({None if self.add_encoders is None else type(self.add_encoders.get('transformer', None))}).",
+                logger,
+            )
+            raise_if_not(
+                same_encoders,
+                f"Encoders loaded from the checkpoint ({tfm_save.add_encoders}) "
+                f"are different from the encoders defined in the new model "
+                f"({self.add_encoders}).",
+                logger,
+            )
+
+            new_add_encoders: Dict = copy.deepcopy(tfm_save.add_encoders)
+            new_encoders: SequentialEncoder = copy.deepcopy(tfm_save.encoders)
+        else:
+            raise_if(
+                len(tfm_save.add_encoders) > 0 and self.add_encoders is None,
+                "Model was created without encoders and encoders were not loaded, but "
+                "the weights were trained using encoders. Either set `load_encoders` to "
+                "`True` or add a matching `add_encoders` dict at model creation.",
```

Comment:
maybe give here the `tfm_save.add_encoders` so that user knows what the required encoders are?